lib/bolt/project.rb

```diff
@@ -45,20 +46,34 @@ def initialize(path, type = 'option')
       @resource_types = @path + '.resource_types'
       @type = type
 
-      project_file = @path + 'project.yaml'
-      @data = Bolt::Util.read_optional_yaml_hash(File.expand_path(project_file), 'project') || {}
+      @project_file = @path + 'project.yaml'
+      @data = Bolt::Util.read_optional_yaml_hash(File.expand_path(@project_file), 'project') || {}
       validate
     end
 
     def to_s
       @path.to_s
     end
 
+    # This API is used to prepend the project as a module to Puppet's internal
+    # module_references list. CHANGE AT YOUR OWN RISK
+    def to_h
+      { path: @path, name: name }
+    end
+
     def eql?(other)
       path == other.path
     end
     alias == eql?
 
+    def file?
+      File.exist?(@project_file)
```

Comment:
Having renamed `boltdir` to `project` muddles this behavior a bit. You now _always_ have a project, but it may or may not be "a project" depending on whether it has `project.yaml`.

Maybe the logic then is that there is _always_ a project, but it simply may or may not be loaded as a module. In that case, we could rename this method to something that expresses exactly what we mean, ie. `load_as_module?`